db.go

```diff
@@ -221,3 +221,46 @@ func (db *DB) Tags() (*Tags, error) {
 	setGcClose(tags)
 	return tags, nil
 }
+
+// GetConfigList returns the config list, which can be used to iterate over all
+// set options starting with prefix.
+func (db *DB) GetConfigList(prefix string) (*ConfigList, error) {
+	cstr := C.CString(prefix)
+	defer C.free(unsafe.Pointer(cstr))
+
+	var ccl *C.notmuch_config_list_t
+	cclptr := (**C.notmuch_config_list_t)(&ccl)
+	err := statusErr(C.notmuch_database_get_config_list(db.toC(), cstr, cclptr))
+	if err != nil {
+		return nil, err
+	}
+	cl := &ConfigList{
+		cptr:   unsafe.Pointer(ccl),
+		parent: (*cStruct)(db),
+	}
+	setGcClose(cl)
+	return cl, nil
+}
+
+// GetConfig gets config value of key
+func (db *DB) GetConfig(key string) (string, error) {
+	ckey := C.CString(key)
+	defer C.free(unsafe.Pointer(ckey))
+	var cval *C.char
+	defer C.free(unsafe.Pointer(cval))
```

Comment:
This needs to go after the error check, or you'll try to free a null pointer if the error is non-nil.

Reply:
sorry for that...